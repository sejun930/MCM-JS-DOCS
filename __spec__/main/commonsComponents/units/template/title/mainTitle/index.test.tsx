import { render } from "@testing-library/react";
import { RecoilRoot } from "recoil";

import _MainTitleTemplate from "src/main/commonsComponents/units/template/title/mainTitle";

export default describe("mainTitle Template Page", () => {
  // 스냅샷
  test("Snapshot", () => {
    const { container } = render(
      <RecoilRoot>
        <_MainTitleTemplate />
      </RecoilRoot>
    );
    expect(container).toMatchSnapshot();
  });

  // h1 태그 존재 여부 확인
  test("Check have h1 and p tag", () => {
    const { container } = render(
      <RecoilRoot>
        <_MainTitleTemplate />
      </RecoilRoot>
    );

    // h1, p 태그를 감싸는 wrapper 태그 존재여부 검증
    const wrapperEle = container.querySelector(".main-title-wrapper");
    expect(wrapperEle).toBeInTheDocument();

    if (wrapperEle) {
      // h1 태그 있는지 검증
      const h1Ele = wrapperEle.querySelector(".main-title");
      expect(h1Ele).toBeInTheDocument();

      // p 태그 있는지 검증
      const pEle = wrapperEle.querySelector(".main-title-remarks");
      expect(pEle).toBeInTheDocument();
    }
  });
});
