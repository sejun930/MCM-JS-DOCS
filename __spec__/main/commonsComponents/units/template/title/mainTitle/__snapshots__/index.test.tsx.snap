// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mainTitle Template Page Snapshot 1`] = `
<div>
  <div
    class="main-title-wrapper css-qe9630"
  >
    <h1
      class="mcm-title-unit main-title"
    >
      ðŸ“– 
    </h1>
    <p
      class="mcm-p-unit main-title-remarks css-z401r8"
    />
  </div>
</div>
`;
